plugins/woocommerce/src/Internal/Orders/OrderActionsRestController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\Orders;
+
+use Automattic\WooCommerce\Internal\Traits\AccessiblePrivateMethods;
+use Automattic\WooCommerce\Internal\RestApiControllerBase;
+use WP_Error;
+use WP_REST_Request;
+
+/**
+ * Controller for the REST endpoint to run actions on orders.
+ *
+ * This first version only supports sending the order details to the customer (`send_order_details`).
+ */
+class OrderActionsRestController extends RestApiControllerBase {
+	use AccessiblePrivateMethods;
+
+	/**
+	 * Get the WooCommerce REST API namespace for the class.
+	 *
+	 * @return string
+	 */
+	protected function get_rest_api_namespace(): string {
+		return 'order-actions';
+	}
+
+	/**
+	 * Register the REST API endpoints handled by this controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->route_namespace,
+			'/orders/(?P<id>[\d]+)/actions/send_order_details',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => fn( $request ) => $this->run( $request, 'send_order_details' ),
+					'permission_callback' => fn( $request ) => $this->check_permissions( $request ),
+					'args'                => $this->get_args_for_order_actions(),
+					'schema'              => $this->get_schema_for_order_actions(),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Permission check for REST API endpoint.
+	 *
+	 * @param WP_REST_Request $request The request for which the permission is checked.
+	 * @return bool|WP_Error True if the current user has the capability, otherwise a WP_Error object.
+	 */
+	private function check_permissions( WP_REST_Request $request ) {
+		$order_id = $request->get_param( 'id' );
+		$order    = wc_get_order( $order_id );
+
+		if ( ! $order ) {
+			return new WP_Error( 'woocommerce_rest_not_found', __( 'Order not found', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		return $this->check_permission( $request, 'read_shop_order', $order_id );
```

Comment:
🤔 Is this the right capability to check here? I would expect it to be a higher level, like `edit_shop_order`. But perhaps there is precedent for using this particular capability?

Reply:
I guess you have to a shop manager or higher to even read orders, so this is probably fine.